src/Symfony/Bundle/FrameworkBundle/CacheWarmer/ValidatorCacheWarmer.php

```diff
@@ -77,14 +78,14 @@ protected function warmUpPhpArrayAdapter(PhpArrayAdapter $phpArrayAdapter, array
     /**
      * @param LoaderInterface[] $loaders
      *
-     * @return XmlFileLoader[]|YamlFileLoader[]
+     * @return list<XmlFileLoader|YamlFileLoader|AttributeLoader>
      */
     private function extractSupportedLoaders(array $loaders): array
     {
         $supportedLoaders = [];
 
         foreach ($loaders as $loader) {
-            if ($loader instanceof XmlFileLoader || $loader instanceof YamlFileLoader) {
+            if (method_exists($loader, 'getMappedClasses')) {
                 $supportedLoaders[] = $loader;
             } elseif ($loader instanceof LoaderChain) {
                 $supportedLoaders = array_merge($supportedLoaders, $this->extractSupportedLoaders($loader->getLoaders()));
```

Comment:
The `LoaderChain` can implement the `getMappedClasses` method, so that we can remove this specific case in 8.0, once validator 7.4+ is required.

Reply:
(also, the existing code would still be needed to keep support for older versions, so in the end nothing will be simplified. It could be in the future, but again all this is mostly internal, not worth more IMHO)